packages/dev-console-standalone/src/UIExtensionsDevTool.tsx

```diff
@@ -0,0 +1,194 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import {
+  ChevronRightMinor,
+  RefreshMinor,
+  ToolsMajor,
+  ViewMinor,
+  HideMinor,
+  MobileMajor,
+} from '@shopify/polaris-icons';
+import {useI18n} from '@shopify/react-i18n';
+import {ExtensionPayload} from '@shopify/ui-extensions-dev-console';
+import {useDevConsoleInternal} from '@/hooks/useDevConsoleInternal';
+import {ToastProvider} from '@/hooks/useToast';
+
+import {Checkbox} from './CheckBox';
+import {ExtensionRow} from './ExtensionRow';
+import {Action} from './ActionSet/Action';
+import * as styles from './UIExtensionsDevTool.module.scss';
+import * as actionSetStyles from './ActionSet/ActionSet.module.scss';
+import en from './translations/en.json';
+
+// Hiding content until there are more options in the side nav
+const DISPLAY_SIDENAV = false;
+
+function getUuid({uuid}: {uuid: string}) {
+  return uuid;
+}
+
+export function UIExtensionsDevTool() {
+  const [i18n] = useI18n({
+    id: 'UIExtensionsDevTool',
+    fallback: en,
+  });
+  const [selectedExtensionsSet, setSelectedExtensionsSet] = useState<Set<string>>(new Set());
+  const {
+    state: {extensions},
+    refresh,
+    show,
+    hide,
+    focus,
+    unfocus,
+  } = useDevConsoleInternal();
+
+  const allSelected = selectedExtensionsSet.size === extensions.length;
+
+  const toggleSelectAll = () => {
+    if (allSelected) {
+      setSelectedExtensionsSet(new Set([]));
+    } else {
+      setSelectedExtensionsSet(new Set(extensions.map(getUuid)));
+    }
+  };
+
+  const toggleSelect = useCallback((extension: ExtensionPayload) => {
+    setSelectedExtensionsSet((set) => {
+      // if delete === false, extension not selected; therefore select it instead
+      if (!set.delete(extension.uuid)) set.add(extension.uuid);
+      return new Set(set);
+    });
+  }, []);
+
+  const selectedExtensions = useMemo(
+    () => extensions.filter((extension) => selectedExtensionsSet.has(extension.uuid)),
+    [extensions, selectedExtensionsSet],
+  );
+
+  const refreshSelectedExtensions = () => refresh(selectedExtensions);
+
+  const [activeMobileQRCodeExtension, setActiveMobileQRCodeExtension] =
+    useState<ExtensionPayload>();
+
+  const actionHeaderMarkup = useMemo(() => {
```

Comment:
Not for this PR, but maybe there is an opportunity to make this its own component, when we start to break things up more.